designsystem/src/commonMain/kotlin/org/hisp/dhis/mobileui/designsystem/component/Button.kt

```diff
@@ -0,0 +1,170 @@
+            val textColor = if (enabled) SurfaceColor.Primary else TextColor.OnDisabledSurface
+
+            ElevatedButton(
+                onClick = { onClick() },
+                modifier = Modifier,
+                elevation = ButtonDefaults.elevatedButtonElevation(),
+                enabled = enabled,
+                colors = ButtonDefaults.filledTonalButtonColors(SurfaceColor.ContainerLow, SurfaceColor.Primary, SurfaceColor.DisabledSurface, TextColor.OnDisabledSurface),
+                shape = ButtonDefaults.outlinedShape,
+                contentPadding = paddingValues
+            ) {
+                ButtonText(text, textColor, icon)
+            }
+        }
+        ButtonStyle.TONAL -> {
+            val textColor = if (enabled) TextColor.OnPrimaryContainer else TextColor.OnDisabledSurface
+            CompositionLocalProvider(LocalRippleTheme provides Ripple.CustomDHISRippleTheme) {
+                SimpleButton(
+                    onClick = { onClick() },
+                    modifier = Modifier,
+                    enabled = enabled,
+                    buttonColors = ButtonDefaults.filledTonalButtonColors(SurfaceColor.PrimaryContainer, TextColor.OnPrimaryContainer, SurfaceColor.DisabledSurface, TextColor.OnDisabledSurface),
+                    text = text,
+                    textColor = textColor,
+                    icon = icon
+                )
+            }
+        }
+        ButtonStyle.KEYBOARDKEY -> {
+            val textColor = if (enabled) SurfaceColor.Primary else TextColor.OnDisabledSurface
+            val shadowColor = if (enabled) SurfaceColor.ContainerHighest else Color.Transparent
+
+            SimpleButton(
+                onClick = { onClick() },
+                modifier = Modifier.buttonShadow(shadowColor, Radius.Full, icon != null),
+                enabled = enabled,
+                buttonColors = ButtonDefaults.filledTonalButtonColors(SurfaceColor.Container, SurfaceColor.Primary, SurfaceColor.DisabledSurface, TextColor.OnDisabledSurface),
+                text = text,
+                textColor = textColor,
+                icon = icon
+            )
+        }
+        ButtonStyle.OUTLINED -> {
+            val textColor = if (enabled) SurfaceColor.Primary else TextColor.OnDisabledSurface
+            OutlinedButton(
+                onClick = { onClick() },
+                enabled = enabled,
+                shape = ButtonDefaults.outlinedShape,
+                colors = ButtonDefaults.outlinedButtonColors(Color.Transparent, SurfaceColor.Primary, Color.Transparent, TextColor.OnDisabledSurface),
+                border = BorderStroke(1.dp, Outline.Dark),
+                contentPadding = paddingValues
+            ) {
+                ButtonText(text, textColor, icon)
+            }
+        }
+    }
+}
+
+@Composable
+fun SimpleButton(
+    enabled: Boolean = true,
+    buttonColors: ButtonColors,
+    text: String,
+    textColor: Color,
+    modifier: Modifier,
```

Comment:
if you add a default value to modifier like `modifier: Modifier = Modifier,` you can avoid passing it as paramenter in all the methdod calls except KEYOBARDKEY case I think.

Reply:
Resolved! thank you!